src/miranda/convert/eccc_rdrs.py

```diff
@@ -114,19 +115,13 @@ def convert_rdrs(
 
         if len(ncfiles) >= 28:
             for nc in ncfiles:
-                for eng in engine:
-                    try:
-                        ds1 = xr.open_dataset(nc, chunks="auto", engine=eng)
-                        break
-                    except Exception as e:
-                        msg = f"Failed to open {nc} with engine {eng}: {e}"
-                        logging.warning(msg)
-                        if eng == engine[-1]:
-                            msg = f"All engines failed for {nc}"
-                            logging.error(msg)
-                            raise RuntimeError(
-                                f"Failed to open {nc} with both engines: {e}"
-                            ) from e
+                eng = get_engine(nc)
+                try:
+                    ds1 = xr.open_dataset(nc, chunks="auto", engine=eng)
+                except Exception as e:
```

Comment:
It's unfortunate that they don't list what errors are possible, but I think we should specify the exceptions raised here. `Exception` catches every possible exception in Python, including sending `ctrl-c` to the kernel.